module/localisation/BallLocalisation/src/BallModel.h

```diff
@@ -37,37 +43,64 @@ namespace localisation {
 
         static constexpr size_t size = 2;
 
+        using StateVec = Eigen::Matrix<Scalar, size, 1>;
+        using StateMat = Eigen::Matrix<Scalar, size, size>;
+
         struct MeasurementType {
             struct BALL {};
         };
 
-        arma::vec2 processNoiseDiagonal;
+        Eigen::Vector2d processNoiseDiagonal;
```

Comment:
Replace all `VectorXd` with `Matrix<Scalar, X, 1>` and `MatrixXd` with `Matrix<Scalar, X, X>`

Reply:
roger that